buildifier/utils/flags.go

```diff
@@ -24,22 +26,22 @@ func ValidateModes(inputType, mode, lint *string, dflag *bool) error {
 	}
 
 	// Check mode.
-	switch *mode {
-	case "":
-		*mode = "fix"
-
-	case "check", "diff", "fix", "print_if_changed":
-		// ok
+	validModes := []string{"check", "diff", "fix", "print_if_changed"}
+	validModes = append(validModes, additionalModes...)
 
-	default:
-		return fmt.Errorf("unrecognized mode %s; valid modes are check, diff, fix", *mode)
+	recognizedMode := false
```

Comment:
factoring these sort of linear search into a standalone method returning bool usually reads cleaner.

But why not just make validModes a map[string]bool, and then "append" by looping and adding to the set?

Then here just do 'set' membership?

Reply:
I tried to, but it makes the order of the valid arguments in the error message randomized.